nixos/modules/programs/wayland/uwsm.nix

```diff
@@ -0,0 +1,124 @@
+      binds the wayland compositor into `graphical-session-pre.target`,
+      `graphical-session.target`, `xdg-desktop-autostart.target`.
+
+      This is useful for Wayland compositors like Hyprland, Sway, Wayfire,
+      etc. that do not start these targets and services on their own.
+
+      ::: {.note}
+      You must configure `waylandCompositors` suboptions as well
+      so that UWSM knows which compositors to manage.
+      :::
+
+      If you are having trouble starting a service that depends on
+      `graphical-session.target`, while using a WM, enabling this option
+      might help
+    '';
+    package = lib.mkPackageOption pkgs "uwsm" { };
+
+    waylandCompositors = lib.mkOption {
+      description = ''
+        Configuration for UWSM-managed Wayland Compositors. This
+        creates a desktop entry file which will be used by Display
+        Managers like GDM, to allow starting the UWSM managed session.
+      '';
+      type = lib.types.attrsOf (
+        lib.types.submodule (
+          { ... }:
+          {
+            options = {
+              prettyName = lib.mkOption {
+                type = lib.types.str;
+                description = "The full name of the desktop entry file.";
+                example = "ExampleWaylandCompositor";
+              };
+              comment = lib.mkOption {
+                type = lib.types.str;
+                description = "The comment field of the desktop entry file.";
+                example = "An intelligent dynamic tiling Wayland compositor managed by UWSM.";
+              };
+              binPath = lib.mkOption {
+                type = lib.types.path;
+                description = ''
+                  The wayland-compositor binary path that will be called by UWSM.
+
+                  It is recommended to use the `/run/current-system/sw/bin/` path
+                  instead of `pkgs.<compositor>` to avoid version mismatch of the
+                  compositor used by UWSM and the one installed in the system.
+                '';
+                example = "/run/current-system/sw/bin/ExampleCompositor";
+              };
+            };
+          }
+        )
+      );
+      example = lib.literalExpression ''
+        hyprland = {
+          prettyName = "Hyprland";
+          comment = "Hyprland compositor managed by UWSM";
+          binPath = "/run/current-system/sw/bin/Hyprland";
+        };
+        sway = {
+          prettyName = "Sway";
+          comment = "Sway compositor managed by UWSM";
+          binPath = "/run/current-system/sw/bin/sway";
+        };
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    systemd.packages = [ cfg.package ];
+    environment.pathsToLink = [ "/share/uwsm" ];
+
+    services.graphical-desktop.enable = true;
+
+    # UWSM recommends dbus broker for better compatibility
+    services.dbus.implementation = "broker";
```

Comment:
Could this become an individual option?
In my opinion, it is confusing if this module changes options for other services internally without any mentions in the docs.
Especially since it would probably _still work_ for most use cases with the `dbus` implementation.
But it could still be `broker` by default if `uwsm` is enabled.
Or maybe we could define it with `lib.mkDefault` ?

Reply:
I guess I can mention it in the option description, having another option seems redundant, and people can just do 

```nix
services.dbus.implementation = lib.mkForce "dbus";
```